pkg/action/utils.go

```diff
@@ -52,3 +55,40 @@ func SetNamespace(x interface{}, chart *chart.Chart, targetNS string, NamespaceF
 		i.Config.SetNamespace(namespace)
 	}
 }
+
+// Name returns the name that should be used based of annotations
+func GetName(chart *chart.Chart, nameTemplate string, args []string) (string, error) {
+	// args here could be: [NAME] [CHART]
+	// cobra flags have been already stripped
+
+	flagsNotSet := func() error {
+		if nameTemplate != "" {
+			return errors.New("cannot set --name-template and also specify a name")
+		}
+		return nil
+	}
+
+	if len(args) > 2 {
```

Comment:
I'm not sure this is going to work out between `hypper install` and `hypper upgrade`. Install accepts 2 arguments, but upgrade only accepts 1, so `GetName()` can be reused.

Reply:
it works because we are only using the args in the case if install. In that case we return on line 75, otherwise (upgrade or install with no name argument) we extract the name from annotations.

We could always end up removing the name param from install and just make it a flag like in upgrade and default to getting the name from the annotations always. Im not sure why upgrade was meant to be changed like that but not install..

Reply:
True! Thanks, now I see it :D.  In the case of upgrade, I suppose it is because in Helm you can't change the release name in upgrade, only upgrade. in install, we support setting a name.

I'm not completely sold on calling the function like `action.GetName(ch, "", make([]string, 0))`. It feels hackish, and people using the lib may not realize that's what it needs, without looking at the function code.

How about leaving install.Name() and upgrade.Name(), and creating an unexported function in action/utils.go to deduplicate the `if` in both?:
```
	if chart.Metadata.Annotations != nil {
		if val, ok := chart.Metadata.Annotations["hypper.cattle.io/release-name"]; ok {
			return val, nil
		}
		if val, ok := chart.Metadata.Annotations["catalog.cattle.io/release-name"]; ok {
			return val, nil
		}
	}
```

Or at the very least, documenting the new GetName() in a way that a lib user doesn't need to necessarily know about args or flags.

Reply:
> I'm not completely sold on calling the function like action.GetName(ch, "", make([]string, 0)). It feels hackish, and people using the lib may not realize that's what it needs, without looking at the function code.

Can't we make this a variadic function so that `action.GetName(ch)` is sufficient in that case ?

Reply:
done!